src/mappings/erc20Templates.ts

```diff
@@ -204,6 +198,25 @@ export function handleNewPaymentCollector(event: NewPaymentCollector): void {
   token.save()
 }
 
+export function handleProviderFee(event: ProviderFee): void {
+  const order = getOrder(
+    event.transaction.hash.toHex(),
```

Comment:
event.transaction.hash could be reuseOrder transaction, if that's the case then won't be able to load Order entity directly?

not sure if providerFee event emitted during reuseOrder store under OrderReuse entity
OR
event.transaction.hash => get OrderReuse => getOrder => store under Order entity

Reply:
It's ready for another review, I've reworked this + added tests for both the order and reuseOrder scenarios